### PR TITLE
Open-Source FRC Viewer Improvements 

### PR DESCRIPTION
**Pull request by austinmatherne-wk**
#### Reason for change
Over the past few months a number of improvements have been made to the viewer for the FRC. These improvements have been developed on a separate fork, but are now ready to be merged back into the open-source Arelle repo.

#### Description of change
* WCAG accessibility compliance:
  * Keyboard navigation
  * Screen reader support
* Tool tips describing key XBRL concepts
* More XBRL details displayed in inspector panel
* Dark mode
* Translations:
  * New Walsh translation
  * More content translated to Spanish
  * Menu to switch UI language between translated languages
* User settings persist in local storage
* More opt-in features for deploying within a larger app:
  * Help guide link
  * Survey link
  * Home link
* Runtime features config file

#### Steps to Test
* CI

**review**:
@Arelle/arelle
@paulwarren-wk

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(2) Security relevant changes were detected</summary>
<li>Watched keyword <code>.html(f</code> in <code>iXBRLViewerPlugin/viewer/src/js/ixbrlviewer.js</code> line(s) ['168'] added</li>
<li>Watched keyword <code>style=</code> in <code>iXBRLViewerPlugin/viewer/src/html/inspector.html</code> line(s) ['184', '198', '212', '227', '241', '255', '269', '285', '297', '340'] added</li>
</details>

## Action Items
* Obtain a [security review](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); reviewer should pay special attention to insights listed above
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: iXBRLViewerPlugin/constants.py [hunk 1]
```diff
@@ -23,13 +23,98 @@
...
+        cliAction='store_true',
+        cliDefault=None,
+        guiVisible=False,
+        guiDefault=None
     )
 ]
-
 GUI_FEATURE_CONFIGS = [c for c in FEATURE_CONFIGS if c.guiVisible]
+
+MANDATORY_FACTS = {
```
Comment (brettkail-wk) [inline iXBRLViewerPlugin/constants.py:101]: It seems surprising to embed this metadata here.

**Review by brettkail-wk — APPROVED** [review]

**Review by derekgengenbacher-wf — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.

File: iXBRLViewerPlugin/constants.py (hunk 1 above)
Comment (paulwarren-wk) [reply iXBRLViewerPlugin/constants.py:101]: Raised https://github.com/Arelle/ixbrl-viewer/issues/767

**Pull request merged**